code/game/objects/structures/watercloset.dm

```diff
@@ -126,9 +128,49 @@
 			GM.adjustBruteLoss(8)
 		else
 			user << "<span class='warning'>You need a tighter grip!</span>"
+
+	else if(exposed)
+		if(!hiddenitem)
+			user << "<span class='notice'>There is nothing in the drain holder.</span>"
+		else
+			var/obj/item/I = hiddenitem
+			if(ishuman(user))
+				user.put_in_hands(I)
+			else
+				I.loc = get_turf(src)
+			user << "<span class='notice'>You fish [I] out of the drain enclosure.</span>"
 	else
 		..()
 
+/obj/structure/urinal/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/weapon/screwdriver))
+		user << "<span class='notice'>You start to [exposed ? "screw the cap back into place" : "unscrew the cap to the drain protector"]...</span>"
+		playsound(loc, 'sound/effects/stonedoor_openclose.ogg', 50, 1)
+		if(do_after(user, 20/I.toolspeed, target = src))
+			user.visible_message("[user] [exposed ? "screws the cap back into place" : "unscrew the cap to the drain protector"]!", "<span class='notice'>You [exposed ? "screw the cap back into place" : "unscrew the cap on the drain"]!</span>", "<span class='italics'>You hear metal and squishing noises.</span>")
+			exposed = !exposed
+	else if(exposed)
+		if (hiddenitem)
+			user << "<span class='warning'>There is already something in the drain enclosure.</span>"
+			return
+		if(I.w_class > 1)
+			user << "<span class='warning'>[I] is too large for the drain enclosure.</span>"
+			return
+		if(!user.drop_item())
+			user << "<span class='warning'>\The [I] is stuck to your hand, you cannot put it in the drain enclosure!</span>"
```

Comment:
since your using `[i]` here, you won't need `\` or `The`...just `[I] is stuck. . .` will suffice.